### ISSUE TITLE
Cannot access click event properties

### ISSUE THREAD

**Issue by mwilde345** [bkrem/react-d3-tree#168]
Thank you for taking the time to report an issue with react-d3-tree!  

Feel free to delete any questions that do not apply.

## Are you reporting a bug, or opening a feature request?
Bug
  
## What is the actual behavior/output?
When clicking a node, the onClick property function is run, as expected. But the event object passed to it gets nullified before I can access any properties. Even after I call event.persist();

## What is the behavior/output you expect?
I want to detect if a user is holding the shift key while clicking a node. Sounds simple enough.
I want the event object to not be null by the time I try and access it. 

## Can you consistently reproduce the issue/create a reproduction case (e.g. on https://codesandbox.io)?
I asked this on stack overflow: https://stackoverflow.com/questions/53643198/react-synthetic-event-strange-behavior/53646853#53646853
and the answer suggests I add the `persist()` call on the events in this library. Is that correct?

you can try it on codesandbox: https://codesandbox.io/s/w7qz6z3rjk
But the click events always give an undefined event object here.
So I created a barebones repo: https://github.com/mwilde345/reactBrokenClick

## What version of react-d3-tree are you using?
latest

**Comment by bkrem** [comment]
Fixed in [v1.12.1](https://github.com/bkrem/react-d3-tree/releases/tag/v1.12.1).

Thanks for pointing this out, this bug has clearly been hiding in the implementation for a while. 

Weirdly I couldn't reproduce it the way it was on the codesandbox (and your example repo), I only managed to reproduce it myself by mocking asynchronous behaviour:

```js
  handleClick(d,evt) {
    setTimeout(() => console.log(evt), 1000);
  }
```

Closing the issue but feel free to reopen if this still seems to be an issue.

**Comment by mwilde345** [comment]
wow! Thanks for getting to this so quickly, it works a charm.
